build_runner/lib/src/entrypoint/base_command.dart

```diff
@@ -102,6 +102,7 @@ abstract class BuildRunnerCommand extends Command<int> {
         delayWritesOption,
         help: 'Delays all file system interactions until the end of a build, '
             'potentially reducing load on tools like the analysis server.',
+        defaultsTo: null,
```

Comment:
Interesting I didn't know null was allowed for flags :). It feels weird, but does make sense here I think. Easier than manually checking `wasParsed` just to set the value back to null haha.